backend/src/main/java/saci/domain/service/KnowledgeService.java

```diff
@@ -0,0 +1,30 @@
+package saci.domain.service;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+import saci.domain.model.Knowledge;
+import saci.domain.service.exceptions.AlreadyExistsException;
+import saci.infrastructure.KnowledgeRepository;
+
+@Service
+public class KnowledgeService {
+
+    private final KnowledgeRepository knowledgeRepository;
+
+    public KnowledgeService(KnowledgeRepository knowledgeRepository) {
```

Comment:
Instead of using a constructor you can use an annotation at the class level, @RequiredArgsConstructor or @AllArgsConstructor, see which one suits it better.